components/elm/src/biogeophys/SnowSnicarMod.F90

```diff
@@ -2238,6 +2393,86 @@ subroutine SNICAR_AD_RT (flg_snw_ice, bounds, num_nourbanc, filter_nourbanc,  &
                          ext_cff_mss_snw_lcl(i) = ext_cff_mss_snw_dfs(rds_idx,bnd_idx)
                       enddo
                    endif
+				   
+                  ! Calculate the asymetry factors under different snow grain shapes
+                   do i=snl_top,snl_btm,1
+                      if(snw_shp_lcl(i) == 2) then ! spheroid
+                         diam_ice = 2._r8*snw_rds_lcl(i)
+                         if(snw_fs_lcl(i) == 0) then
```

Comment:
Change `0` --> `0._r8` here and all other locations for `snw_fs_lcl`

Reply:
Done. Made same change for snw_ar_lcl